component/core.py

```diff
@@ -389,16 +433,12 @@ def component(self, usage=None, model_name=None):
         elif len(component_classes) > 1:
             # If we have more than one component, try to find the one
             # specifically linked to the collection...
-            component_classes = [
-                c for c in component_classes if c._collection == self.collection._name
-            ]
+            component_classes = self._filter_components_by_collection(component_classes)
```

Comment:
@guewen seems to me that this filtering on collection and model it's done twice because `registry.lookup` already does it. Am I wrong?

Reply:
`registry.lookup` filters on the same collection or no collection. Here, we narrow down to a single component matching the collection if we have one component on the collection and one without.